crates/core/src/node/inner/vm_runner.rs

```diff
@@ -199,15 +194,62 @@ impl VmRunner {
         let spent_on_pubdata =
             tx_result.statistics.gas_used - tx_result.statistics.computational_gas_used as u64;
 
-        let status = match &tx_result.result {
-            ExecutionResult::Success { .. } => "SUCCESS",
-            ExecutionResult::Revert { .. } => "FAILED",
-            ExecutionResult::Halt { .. } => "HALTED",
-        };
+        let mut known_addresses = known_addresses_after_transaction(tx);
+        let mut trace_output = None;
+
+        if !call_traces.is_empty() {
+            let mut builder = CallTraceDecoderBuilder::default();
+
+            builder = builder.with_signature_identifier(
+                SignaturesIdentifier::new(Some(config.get_cache_dir().into()), config.offline)
+                    .map_err(|err| {
+                        anyhow::anyhow!("Failed to create SignaturesIdentifier: {:#}", err)
+                    })?,
+            );
+
+            let decoder = builder.build();
+            let mut arena = build_call_trace_arena(&call_traces, &tx_result);
+            decode_trace_arena(&mut arena, &decoder).await?;
+
+            extract_addresses(&arena, &mut known_addresses);
+
+            let verbosity = get_shell().verbosity;
+            if verbosity >= 2 {
+                let filtered_arena = filter_call_trace_arena(&arena, verbosity);
+                trace_output = Some(render_trace_arena_inner(&filtered_arena, false));
+            }
+            if !config.disable_console_log {
+                self.console_log_handler
+                    .handle_calls_recursive(&call_traces);
+            }
+        }
+
+        let balance_diffs: Vec<formatter::transaction::BalanceDiff> =
+            extract_balance_diffs(&known_addresses, &tx_result.logs.storage_logs)
+                .into_iter()
+                .map(Into::into)
+                .collect();
 
         // Print transaction summary
         if config.show_tx_summary {
-            formatter::print_transaction_summary(config.get_l2_gas_price(), tx, &tx_result, status);
+            sh_eprintln!(
+                "{}",
+                formatter::transaction::TransactionSummary::new(
+                    config.get_l2_gas_price(),
+                    tx,
+                    &tx_result,
+                    balance_diffs,
+                )
+            );
+        }
+
+        if let Some(trace_output) = trace_output {
+            sh_println!("\nTraces:\n{}", trace_output);
```

Comment:
Am I missing something or do we print traces twice? (duplicate in comment below)

Reply:
Thanks!